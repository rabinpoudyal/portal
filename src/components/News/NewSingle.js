import React from "react";
import styled from "styled-components";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

const NewSingle = ({ item }) => (
  <Card>
    <CardActionArea>
      <CardMedia
        image={item.urlToImage}
        style={{ height: 0, paddingTop: "56%" }}
        title={item.source.name}
      />
      <CardContent>
        <EqualHeight>
          <Typography
            gutterBottom
            variant="h5"
            component="h2"
            className="card-title"
          >
            {item.title}
          </Typography>
          <hr />
          <Typography component="p">{item.description}</Typography>
        </EqualHeight>
        <Typography component="p" color="primary">
          Source: {item.source.name}
        </Typography>
      </CardContent>
    </CardActionArea>
    <CardActions>
      <Button size="small" color="secondary" href={item.url} target="_blank">
        Full Article
      </Button>
      <Button size="small" color="secondary" href={`/news/${item.title}`}>
        Learn More
      </Button>
    </CardActions>
  </Card>
);

const EqualHeight = styled.div`
  height: 230px;
`;

export default NewSingle;
