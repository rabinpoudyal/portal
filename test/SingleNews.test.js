import NewSingle from "../src/components/News/NewSingle";
import { Typography } from "@material-ui/core";
import { exportAllDeclaration } from "@babel/types";

const newsItem = {
  urlToImage: "google.com",
  title: "This is the title",
  source: {
    name: "google news"
  },
  description: "This is the news description"
};

describe("<NewSingle/>", () => {
  it("accepts the item props correctly", () => {
    //console.log(newsItem);
    const wrapper = mount(<NewSingle item={newsItem} />);
    console.log(wrapper.props().item);
    //console.log(newsItem.title);
    //expect(wrapper.find(".card-title").text()).to.equal(newsItem.title);
  });
});
