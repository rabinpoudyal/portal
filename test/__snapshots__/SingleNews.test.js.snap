// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewSingle/> accepts the item props correctly 1`] = `ShallowWrapper {}`;
